File: Leet_Code/Valid_Parentheses.py
class Solution(object):
    def isValid(self, s):
        closes = [')','}',"]"]
        opens = ['(','{',"["]
        stack = []
        for i in range(len(s)):
            if stack == []:
                if s[i] in closes:
                    return False
                else:
                    stack.append(s[i])
            else:
                if s[i] in closes:
                    if opens.index(stack[-1]) == closes.index(s[i]):
                        stack.pop()
                    else:
                        return False
                else:
                    stack.append(s[i])
        if stack == []:
            return True
        else:
            return False

File: Leet_Code/Valid_Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        vector<char> ans;
        for(int i = 0; i < s.size(); i++) {
            if (s[i] == '(' || s[i] == '{' || s[i] == '[') {
                ans.push_back(s[i]);
            }
            else {
                if (!ans.size()) return false;
                char q = ans.back();
                ans.pop_back();
                if (s[i] == ')' && q != '(') return false;
                else if (s[i] == '}' && q != '{') return false;
                else if (s[i] == ']' && q != '[') return false;
            }
            
        } return !ans.size();
    }
};